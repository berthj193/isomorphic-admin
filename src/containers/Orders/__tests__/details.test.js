import React from 'react';
import OrderDetails, { OrderDetails as RawOrderDetails } from '../details';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

describe('<OrderDetails />', () => {
  const defaultParams = {
    match: {
      params: {},
    },
    orders: [{}],
  };
  const getComponent = (props = {}) => shallow(<RawOrderDetails {...defaultParams} {...props} />);
  const mockStore = configureStore();

  it('should render properly', () => {
    expect(toJson(getComponent({ orders: [{ id: 1 }] }))).toMatchSnapshot();
  });

  it('should render loading page', () => {
    expect(toJson(getComponent())).toMatchSnapshot();
  });

  it('should render details with provider', () => {
    expect(toJson(shallow(<OrderDetails store={mockStore()} />))).toMatchSnapshot();
  });
});
