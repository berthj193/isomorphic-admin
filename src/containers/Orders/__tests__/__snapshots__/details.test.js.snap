// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderDetails /> should render details with provider 1`] = `
<ComponentWithProvider
  startRequest={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;

exports[`<OrderDetails /> should render loading page 1`] = `
<PageLoading
  style={
    Object {
      "height": "50vh",
    }
  }
/>
`;

exports[`<OrderDetails /> should render properly 1`] = `
<React.Fragment>
  <LayoutWrapper
    style={
      Object {
        "minHeight": "100vh",
      }
    }
  >
    <LayoutContent>
      <h1>
        Orders
      </h1>
      <ButtonList
        buttons={
          Array [
            Object {
              "link": "/dashboard/orders",
              "title": "Go back",
            },
            Object {
              "link": "/dashboard/orders/undefined/refund",
              "title": "Add refund",
            },
          ]
        }
      />
      <PageDetails
        details={
          Array [
            Object {
              "Store": undefined,
            },
            "---",
            Object {
              "E-mail": undefined,
            },
            Object {
              "First name": undefined,
            },
            Object {
              "Last name": undefined,
            },
            Object {
              "Phone number": undefined,
            },
            "---",
            Object {
              "First name": undefined,
            },
            Object {
              "Last name": undefined,
            },
            Object {
              "City": undefined,
            },
            Object {
              "Region": undefined,
            },
            Object {
              "Postcode": undefined,
            },
            Object {
              "Line one": undefined,
            },
            Object {
              "Line two": undefined,
            },
            "---",
            Object {
              "Item total": "undefined undefined",
            },
            Object {
              "Grand total": "undefined undefined",
            },
            Object {
              "Shipping total": "undefined undefined",
            },
            Object {
              "Item discount total": "undefined undefined",
            },
            Object {
              "Shipping discount total": "undefined undefined",
            },
          ]
        }
      />
    </LayoutContent>
  </LayoutWrapper>
</React.Fragment>
`;
